### PR TITLE
Readme Update with pre-commit Pipeline

### PR DESCRIPTION
**Pull request by AndreSlavescu**
Added complete pipeline to format files before contributing to repository. 

automatically runs black on all files for formatting. 
automatically runs isort on all files for alphabetical ordering.

### CODE REVIEW

**Comment by ronakice** [comment]
I believe changes also need to be made to pyproject.toml?

File: requirements.txt [hunk 1]
```diff
@@ -4,8 +4,6 @@ tiktoken>=0.5.2
 transformers>=4.37.0
 pyserini>=0.24.0
 python-dotenv>=1.0.1
-faiss-gpu>=1.7.2
```
Comment (sahel-sh) [inline requirements.txt:7]: This is needed, I remember I had issues without it. please keep it

File: pyproject.toml [hunk 2]
```diff
@@ -28,10 +28,8 @@ dependencies = [
     "pyserini == 0.24.0",
     "python-dotenv == 1.0.1",
     "faiss-cpu == 1.7.2",
-    "faiss-gpu == 1.7.2",
```
Comment (sahel-sh) [inline pyproject.toml:31]: same here, let's not change requirements

**Review by sahel-sh — CHANGES_REQUESTED** [review]

**Comment by sahel-sh** [comment]
As we discussed in the chat, let's keep requirement changes separate. Ronak has already created an issue for it: https://github.com/castorini/rank_llm/issues/66

**Comment by ronakice** [comment]
@AndreSlavescu can we go about this as Sahel suggested? Sahel has a separate PR addressing this

**Pull request closed without merging**